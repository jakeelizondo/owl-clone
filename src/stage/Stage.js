import React from 'react';
import Participant from '../participant/Participant.js';
import './Stage.css';

function Stage(props) {
  let participantsFiltered = props.participants.filter(
    (person) => person.onStage === true
  );

  let participantComps = participantsFiltered.map((participant) => {
    return (
      <Participant
        key={participant.id}
        avatar={participant.avatar}
        id={participant.id}
        inSession={participant.inSession}
        name={participant.name}
        invertImage={true}
      />
    );
  });

  console.log(participantComps);

  return <div className="on-stage-person">{participantComps}</div>;
}

export default Stage;
