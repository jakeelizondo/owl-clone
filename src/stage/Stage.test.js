import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage';
import renderer from 'react-test-renderer';
import participants from '../participants';

describe('Stage Component', () => {
  // smoke test for stage component

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage participants={participants} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Stage participants={participants} key={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
