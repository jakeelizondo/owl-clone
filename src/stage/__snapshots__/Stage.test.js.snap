// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage Component renders the UI as expected 1`] = `
<div
  className="on-stage-person"
>
  <div
    className="participant"
  >
    <div>
      <h2>
        Koren Templeton
      </h2>
      <img
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
  <div
    className="participant"
  >
    <div>
      <h2>
        Ashla Attwool
      </h2>
      <img
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
</div>
`;
